lib/createGltf.js

```diff
@@ -237,11 +237,10 @@ function addTexture(gltf, texture) {
 
 function getTexture(gltf, texture) {
     let textureIndex;
-    const name = texture.name;
-    const textures = gltf.textures;
-    const length = textures.length;
+    const images = gltf.images;
+    const length = images.length;
     for (let i = 0; i < length; ++i) {
-        if (textures[i].name === name) {
+        if (images[i].extras._obj2gltf === texture) {
```

Comment:
So this will now duplicate any texture if you're not passing in the same reference, right? (Because even if `texture1` and `texture2` are identical, if they're different objects in memory then `texture1 === texture2` is false).

Reply:
Yup. The previous check was not careful enough because both diffuse and metallic-roughness had the same name but were different texture instances.